### ISSUE TITLE
Getting 'no-undef' warnings when using decorators.

### ISSUE THREAD

**Issue by martndemus** [babel/babel-eslint#204]
When using eslint-babel, I get `no-undef` warnings when declaring decorators on object properties.

```javascript 
 var Model = DS.Model.extend({
    @attr firstName,
    @attr({ defaultTo: 'blue' }) lastName,
    @hasMany user,
    @belongsTo car
  });
```

```
  14:11  error  "firstName" is not defined  no-undef
  15:34  error  "lastName" is not defined   no-undef
  16:14  error  "user" is not defined       no-undef
  17:16  error  "car" is not defined        no-undef
```

Setup: 
```
$ npm ls | grep eslint
├─┬ ember-cli-qunit-eslint@1.0.0
│ ├─┬ babel-eslint@4.1.4
│ ├─┬ broccoli-lint-eslint@1.1.1
│ │ └─┬ eslint@1.8.0
```

```
Some of the output has been trimmed here.
$ npm ls | grep babel
├─┬ ember-cli-babel@5.1.5
│ ├─┬ broccoli-babel-transpiler@5.4.5
│ │ ├─┬ babel-core@5.8.33
│ ├─┬ babel-eslint@4.1.4
│ │ ├─┬ babel-core@5.8.33
```

**Comment by martndemus** [comment]
Hmm, I'm doubting this is a real bug. It might actually be a valid warning.

Example: 
```javascript
{
  @readOnly foo
}
```

is transpiled by babel to

```javascript
{
  key: 'foo',
  decorators: [readOnly],
  initializer: function initializer() {
    return foo;
  }
}
```

Which looks like it's just like a shorthand property.

```javascript
var a = 1;
var b = 2;

var c = { a, b }

c.a // => 1
c.b // => 2
```
So this would cause a nice subtle bug if we would disable `no-undef` warnings on this.
```javascript
var foo = 'bar'

var object = {
  @readOnly foo
};

object.foo // => 'bar'
```

What do you think? @hzoo

**Comment by hzoo** [comment]
Right those are shorthand props - I guess I assumed those were defined somewhere else or were parameters. If the code snippet is just

```
var Model = DS.Model.extend({
    @attr firstName,
    @attr({ defaultTo: 'blue' }) lastName,
    @hasMany user,
    @belongsTo car
  });
```
Then yeah I guess it should error since it would error even without the decorators

**Comment by devinus** [comment]
Umm, I'm actually not sure what's going on here. Is this a declaration that babel-eslint wont support multiple decorators?

<img width="856" alt="screen shot 2015-12-11 at 1 02 18 am" src="https://cloud.githubusercontent.com/assets/4836/11737841/e0002efe-9fa2-11e5-987e-299a2987ab71.png">

**Comment by hzoo** [comment]
I can't tell if this is the same issue or not based on that code.

**Comment by brancusi** [comment]
@martndemus, could you explain this a bit more. Are you saying using this style of code could cause this subtle bug? I am using [ember-computed-decorators](https://github.com/rwjblue/ember-computed-decorators) plugin extensively.

Thanks

**Comment by AO16** [comment]
I am having the same issue with babel-eslint picking showing an error with no-undef.

**Comment by AO16** [comment]
Found this which seems to do the trick: https://github.com/rwjblue/ember-computed-decorators/issues/59#issuecomment-174417204. Hope that is helpful to someone.

**Comment by rwwagner90** [comment]
It would be ideal for this to work. Couldn't there just be some sort of check with no-undef that would say, if there is a decorator, don't flag as undef?

**Comment by NullVoxPopuli** [comment]
I'm hoping for a solution for this too.

in my code, I have

```es7
//...
@alias('model.property') property,
//...
```
which _defines_ property, rather than uses it.

**Comment by rwwagner90** [comment]
@NullVoxPopuli it seems like the best solution for now is to do `@alias('model.property') property: null`
